File: internal/src/main/java/dev/mvmo/sicklang/internal/object/error/ErrorObject.java
package dev.mvmo.sicklang.internal.object.error;

import dev.mvmo.sicklang.internal.object.ObjectType;
import dev.mvmo.sicklang.internal.object.SickObject;

public record ErrorObject(String message) implements SickObject {

    @Override
    public String inspect() {
        return "ERROR: " + message;
    }

    @Override
    public ObjectType objectType() {
        return ObjectType.ERROR;
    }

    public static ErrorObject newError(String message, Object... args) {
        return new ErrorObject(String.format(message, args));
    }

}

File: internal/src/main/java/dev/mvmo/sicklang/internal/object/error/ErrorObject.kt
package dev.mvmo.sicklang.internal.`object`.error

import dev.mvmo.sicklang.internal.`object`.ObjectType
import dev.mvmo.sicklang.internal.`object`.SickObject

class ErrorObject(val message: String) : SickObject {

    override fun inspect() =
        "ERROR: $message"

    override fun objectType() =
        ObjectType.ERROR

    companion object {
        @JvmStatic
        fun formatted(message: String, vararg args: Any): ErrorObject {
            return ErrorObject(String.format(message, *args))
        }
    }

}